drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-client/src/main/java/org/drools/workbench/screens/scenariosimulation/kogito/client/dmn/KogitoScenarioSimulationBuilder.java

```diff
@@ -0,0 +1,441 @@
+
+/**
+ * Class used to populate <code>ScenarioSimulationModel</code> inside <code>Callback</code>s
+ * <p>
+ * Mostly copied from <code>org.drools.workbench.screens.scenariosimulation.backend.server.util.SimulationSettingsCreationStrategy</code> and its children.
+ * <p>
+ * Implemented to manage the nested callbacks for <b>DMN</b>
+ */
+public class KogitoScenarioSimulationBuilder {
+
+    @Inject
+    private KogitoDMNService dmnTypeService;
+
+    private static FactMappingType convert(final FactModelTree.Type modelTreeType) {
+        switch (modelTreeType) {
+            case INPUT:
+                return GIVEN;
+            case DECISION:
+                return EXPECT;
+            default:
+                throw new IllegalArgumentException("Impossible to map");
+        }
+    }
+
+    /**
+     * Populate the given <code>ScenarioSimulationModel</code> and returns it inn the given <code>callback</code>
+     * @param content
+     * @param type
+     * @param value
+     * @param callback
+     */
+    public void populateScenarioSimulationModel(final ScenarioSimulationModel content, final ScenarioSimulationModel.Type type, final String value, RemoteCallback<String> callback) {
+        switch (type) {
+            case RULE:
+                populateRULE(content, value, callback);
+                break;
+            case DMN:
+                populateDMN(content, value, callback);
+                break;
+            default:
+                throw new IllegalArgumentException("Impossible to map");
+        }
+    }
+
+    private void populateRULE(final ScenarioSimulationModel toPopulate, final String dmoSession, final RemoteCallback<String> callback) {
+        toPopulate.setSimulation(createRULESimulation());
+        toPopulate.setBackground(createBackground());
+        toPopulate.setSettings(createRULESettings(dmoSession));
+        convertScenarioSimulationModel(toPopulate, callback);
+    }
+
+    private void populateDMN(final ScenarioSimulationModel toPopulate, final String dmnFilePath, final RemoteCallback<String> callback) {
+        toPopulate.setBackground(createBackground());
+        populateDMNSimulationAndSettings(toPopulate, dmnFilePath, callback);
+    }
+
+    private void convertScenarioSimulationModel(final ScenarioSimulationModel toConvert, final RemoteCallback<String> callback) {
+        JSIScenarioSimulationModelType jsiScenarioSimulationModelType = ApiJSInteropConverter.getJSIScenarioSimulationModelType(toConvert);
+        final SCESIM scesim = Js.uncheckedCast(JsUtils.newWrappedInstance());
+        JsUtils.setNameOnWrapped(scesim, makeJSINameForSCESIM());
+        JsUtils.setValueOnWrapped(scesim, jsiScenarioSimulationModelType);
+        SCESIMMarshallCallback scesimMarshallCallback = getSCESIMMarshallCallback(callback);
+        SCESIMMainJs.marshall(scesim, "", scesimMarshallCallback);
+    }
+
+    private JSIName makeJSINameForSCESIM() {
```

Comment:
Can you add javadoc to this method? I don't understand why it is needed. Where is `JsUtils`?

Reply:
@danielezonca that class is autogenerated in `-marshaller` module